apps/sms/js/inter_instance_event_dispatcher.js

```diff
@@ -25,33 +31,25 @@
 
   const Dispatcher = EventDispatcher.mixin({
     connect: function() {
-      if (worker) {
+      if (channel) {
         return;
       }
 
-      worker = new SharedWorker('js/iac/shared_worker.js');
-      worker.addEventListener('error', function(e) {
-        console.error('Worker threw an error: ', e);
-      });
-      worker.port.addEventListener('message', onMessage);
-      worker.port.start();
+      channel = new BroadcastChannel('messages-channel');
```

Comment:
nit: maybe we can set 'messages-channel' to const? but it's really minor because it's only used once in real case, so it's up to you.